python/ray/serve/schema.py

```diff
@@ -545,6 +546,34 @@ def get_empty_schema_dict() -> Dict:
         return {"applications": []}
 
 
+@PublicAPI(stability="alpha")
+class ReplicaDetails(BaseModel, extra=Extra.forbid):
+    replica_id: str = Field(
+        description=(
+            "Unique ID for the replica. By default, this will be "
+            '"<deployment name>#<replica suffix>", where the replica suffix is a '
+            "randomly generated unique string."
+        )
+    )
+    state: ReplicaState = Field(description="Current state of the replica.")
+    pid: Optional[int] = Field(description="PID of the replica actor process.")
+    actor_name: str = Field(description="Name of the replica actor.")
+    actor_id: Optional[str] = Field(description="ID of the replica actor.")
+    node_id: Optional[str] = Field(
+        description="ID of the node that the replica actor is running on."
+    )
+    node_ip: Optional[str] = Field(
+        description="IP address of the node that the replica actor is running on."
```

Comment:
Can a node IP change without the node crashing or restarting? I.e. is there any chance this field ever becomes stale?

Reply:
Ah hmm, I don't know ... maybe we can ask someone on the core team? Not sure who to tag here.

Reply:
I'll tag in @edoakes. I just want to make sure we don't have to deal with any issues around staleness here.

Reply:
no, this is static for the lifetime of the node/replica